### DISCUSSION TITLE
Segmentation Fault on importing spaCy

### DISCUSSION THREAD

**Discussion by sameedmsiddiqui** [explosion/spaCy#12780] [Help: Installation] [answered]
Hi,

I'm using running Ubuntu 22.04.2 LTS on WSL on Windows 11. I've tried to install spaCy using both conda (` conda install -c conda-forge spacy`) and pip  (`pip install -U pip setuptools wheel; pip install -U spacy`). Unfortunately, whenever I open up my python (3.10.11, 64 bit) and `import spacy`, python closes with a Segmentation fault with no other error.

I've tried running this on other conda environments, but to no avail. Does anyone have any thoughts? 

Best,
Sameed

**Comment by svlandeg** [comment, marked as answer]
Hi @sameedmsiddiqui! Sorry to hear you've been running into issues.

With `pip`, have you made sure to start from a fresh new environment? That should avoid any conflicts that could lead to the segmentation fault:
```
python -m venv .env
source .env/bin/activate
pip install -U pip setuptools wheel
pip install -U spacy
```

If that doesn't work, can you share the full console output (commands + stack trace)?

**Reply by sameedmsiddiqui** [reply]
Hi,

This is great, this worked. After doing some debugging, it seems that there seems to be some incompatibility with conda? Specifically, the following leads to a seg fault:

```
conda create -n myenv python=3.10 --no-default-packages
conda activate myenv
conda install spacy
```

Replacing line 3 of the above with `pip install spacy` also leads to a segmentation fault as above. Do you have any thoughts on how to debug what in conda might be interfering with spacy, as my usual workflow uses conda environments. 

Thanks!

**Reply by svlandeg** [reply]
Hey @sameedmsiddiqui,

Glad to hear you got it working with `pip` at least.

With `conda`, can you try using conda forge instead? Replace your line 3 with:
```
conda install -c conda-forge spacy
```
Hopefully that works 🤞

**Reply by sameedmsiddiqui** [reply]
Hi,

Interestingly, it seems to have been a Python version problem?

Specifically, when I create a conda environment with python=3.10, spacy importing spacy causes a segmentation fault; when I don't stipulate a python version, there's no segmentation fault:

No seg fault # Python 3.11.4
```
conda create -n tf4
conda activate tf4
conda install -c conda-forge spacy
```

Seg fault # Python 3.10.11
```
conda create -n tf_forge python=3.10 --no-default-packages
conda activate tf4
conda install -c conda-forge spacy
```

That's super interesting to me, but good to know. Thanks for all the help! :)
